### DISCUSSION TITLE
Extract url's open graph details via RPC

### DISCUSSION THREAD

**Discussion by sandbox-apps** [supabase/supabase#2885] [General]
# Feature request

## Is your feature request related to a problem? Please describe.
Currently not using supabase and planning to migrate using it but I just started to think a checkpoint in my head.
Currently our backend implementation has a way to extract the OG details when we pass a link in the API and return those details. Can we also do this in RPC? We don't want to delegate this action in the frontend.

## Describe the solution you'd like
Extract the Thumbnail, Title, Description, type of open graph (article/video) and other fields

**Comment by kiwicopple** [comment]
Hey @sandbox-apps , could you clarify this one a bit further? (perhaps share an example of your current app?)

I assume you're thinking about Supabase Storage right?

**Comment by sandbox-apps** [comment]
https://www.opengraph.xyz/url/https%3A%2F%2Fwww.byeindonesia.com%2F/

Something like this. maybe a function to call inside a plv8 to easily extract url metadata. This shouldn't need supabase storage.

It just extract
- thumbnailUrl
- title
- description
- etc.. (anything available to extract)

So we can save this inside database
So when we query the db we can save something like this

![Uploading Screen Shot 2021-08-12 at 11.04.21 PM.png…]()

**Comment by kiwicopple** [comment]
I'm unsure if this is a feature we'd implement natively - more of a programming challenge for you ... it can definitely be done using either the [pg_net](https://github.com/supabase/pg_net/) or [pgsql-http](https://github.com/pramsey/pgsql-http) extension, however you'd need to parse the og details from the returned text (or call an external service which passes everything back as JSON).

You could definitely do this with plv8, ([SupaScript](https://github.com/burggraf/SupaScript) might help here.

I'm going to convert this to a Discussion. If you end up building something please do share it with the community!
